webui/react/src/themes.ts

```diff
@@ -269,34 +43,297 @@ const stateColorMapping = {
+  stageWeak: undefined,
+  stageOn: undefined,
+  stageOnStrong: undefined,
+  stageOnWeak: undefined,
+  stageBorder: undefined,
+  stageBorderStrong: undefined,
+  stageBorderWeak: undefined,
+  surface: undefined,
+  surfaceStrong: undefined,
+  surfaceWeak: undefined,
+  surfaceOn: undefined,
+  surfaceOnStrong: undefined,
+  surfaceOnWeak: undefined,
+  surfaceBorder: undefined,
+  surfaceBorderStrong: undefined,
+  surfaceBorderWeak: undefined,
+  float: undefined,
+  floatStrong: undefined,
+  floatWeak: undefined,
+  floatOn: undefined,
+  floatOnStrong: undefined,
+  floatOnWeak: undefined,
+  floatBorder: undefined,
+  floatBorderStrong: undefined,
+  floatBorderWeak: undefined,
+
+  // Interactive styles.
+  ix: undefined,
+  ixStrong: undefined,
+  ixWeak: undefined,
+  ixActive: undefined,
+  ixInactive: undefined,
+  ixOn: undefined,
+  ixOnStrong: undefined,
+  ixOnWeak: undefined,
+  ixOnActive: undefined,
+  ixOnInactive: undefined,
+  ixBorder: undefined,
+  ixBorderStrong: undefined,
+  ixBorderWeak: undefined,
+  ixBorderActive: undefined,
+  ixBorderInactive: undefined,
 
-export const defaultThemeId: ThemeId = ThemeId.LightDetermined;
+  // Specialized and unique styles.
+  density: '2',
+  targetFocus: '0px 0px 4px rgba(0, 155, 222, 0.12)',
+  borderRadius: '4px',
+  borderRadiusStrong: '8px',
+  borderRadiusWeak: '2px',
+  strokeWidth: '1px',
+  strokeWidthStrong: '3px',
+  strokeWidthWeak: '0.5px',
+  elevation: undefined,
+  elevationStrong: undefined,
+  elevationWeak: undefined,
+  overlay: undefined,
+  overlayStrong: undefined,
+  overlayWeak: undefined,
+
+  // Status styles.
+  statusActive: 'rgba(0, 155, 222, 1.0)',
+  statusActiveStrong: undefined,
+  statusActiveWeak: undefined,
+  statusActiveOn: 'rgba(255, 255, 255, 1.0)',
+  statusActiveOnStrong: undefined,
+  statusActiveOnWeak: undefined,
+  statusCritical: 'rgba(204, 0, 0, 1.0)',
+  statusCriticalStrong: undefined,
+  statusCriticalWeak: undefined,
+  statusCriticalOn: 'rgba(255, 255, 255, 1.0)',
+  statusCriticalOnStrong: undefined,
+  statusCriticalOnWeak: undefined,
+  statusInactive: 'rgba(102, 102, 102, 1.0)',
+  statusInactiveStrong: undefined,
+  statusInactiveWeak: undefined,
+  statusInactiveOn: 'rgba(255, 255, 255, 1.0)',
+  statusInactiveOnStrong: undefined,
+  statusInactiveOnWeak: undefined,
+  statusPending: 'rgba(102, 102, 204, 1.0)',
+  statusPendingStrong: undefined,
+  statusPendingWeak: undefined,
+  statusPendingOn: 'rgba(255, 255, 255, 1.0)',
+  statusPendingOnStrong: undefined,
+  statusPendingOnWeak: undefined,
+  statusSuccess: 'rgba(0, 153, 0, 1.0)',
+  statusSuccessStrong: undefined,
+  statusSuccessWeak: undefined,
+  statusSuccessOn: 'rgba(255, 255, 255, 1.0)',
+  statusSuccessOnStrong: undefined,
+  statusSuccessOnWeak: undefined,
+  statusWarning: 'rgba(204, 153, 0, 1.0)',
+  statusWarningStrong: undefined,
+  statusWarningWeak: undefined,
+  statusWarningOn: 'rgba(255, 255, 255, 1.0)',
+  statusWarningOnStrong: undefined,
```

Comment:
do we need to individually define these or can we assume that each key here `k` comes with `kWeak` and `kStrong`? can we generate that programmatically as well?

Reply:
well it seems we can't. it wouldn't apply to all keys. what if we apply it to certain value types and patterns

- if the variations are undefined
- if the original key is `\d+px` or `^rgba`

Reply:
that should make it easier to manually go through this list

Reply:
I was looking to see if we can dynamically define them with TypeScript support but I didn't have any luck. There were enough special cases that didn't follow the pattern, that I ended up going this route.

Reply:
discussed async, decided to pursue this in a separate body of work (assuming we can make it work w/ TypeScript, since the issue might be we are unable to handle dynamic record keys when we are defining types from records that have runtime keys being defined)